### ISSUE TITLE
Feature Suggestion: "Suspend Game" mode.

### ISSUE THREAD

**Issue by TsukiZero** [melonDS-emu/melonDS#1512]
The idea is to add a function bound to a shortcut that creates a savestate, prompts the user about it, and then closes MelonDS. Then next time MelonDS is opened, it loads that savestate, deletes it, and the player resumes from where they were.

How come I had this idea? Well, some games already possesses such feature innately, but may have some additional condition (like not suspending when there are enemies around), plus sometimes I actually do this when I'm playing a game legit and need to take a break (I go, create a save, close MelonDS, do whatever, come to MelonDS, load the ROM, load the save, delete the save.)

**Comment by sandwichwater** [comment]
i was actually thinking of something like this last night , more often than not when i wanna play a quick game of kingdom hearts 358/2 days i opt for just taking out my 3ds and playing it there instead of emulating it on melonds , despite melon having upscaled res , i can play with a controller , and multiple other on the fly changes i cant get on the physical console . the simple reason why is that i just have to open up my 3ds and im right where i left off , whereas emulating i have to go through the startup and menu screens just to get into the game , which only takes a minute or so but still , not really efficient when i just wanna play a quick mission or two . only thing my actual 3ds has over melonds is better audio , even with interpolation for some reason but i guess thats a separate issue for a separate issue report

i guess all im trying to say is yea this would be extremely helpful , the use of savestates would definitely give it more of that ds feel

**Comment by nadiaholmquist** [comment]
> only thing my actual 3ds has over melonds is better audio , even with interpolation for some reason but i guess thats a separate issue for a separate issue report

Could you create an issue for this separately? If you're able to record audio samples that could be appreciated too. Try changing the bitrate setting (I should rename it to bit depth) to 16-bit and see if that changes anything.

**Comment by sandwichwater** [comment]
> Could you create an issue for this separately?

yes sure thing , would samples from melonds interpolation , other emulators interpolation , and what it should sound like outside of emulation be good enough ¿

**Comment by sandwichwater** [comment]
created an issue here #1513

**Comment by Nixel1324** [comment]
For booting up a game quickly, just double click the file (.nds) and select "open with melonds" if it asks. It should only ask the first time you try to open a game, and then it works for all games. You can even create shortcuts to the files and give them custom icons.

**Comment by TsukiZero** [comment]
> For booting up a game quickly, just double click the file (.nds) and select "open with melonds" if it asks. It should only ask the first time you try to open a game, and then it works for all games. You can even create shortcuts to the files and give them custom icons.

But wouldn't it be better if there was a function conveniently bound to a shortcut that did the following in order for you:

- Create a temporary save state
- Prompt the user with a message regarding its function (perhaps with an option like "do not tell me this again")
- Close the emulator
- Next time the emulator is loaded, it AUTOMATICALLY loads the last loaded ROM and its temporary save state
- Deletes the temporary save state.

Of course, it MIGHT be possible to automate the last two points via a batch file, or someone more code-savvy could make a script that does exactly that.

**Comment by sandwichwater** [comment]
> For booting up a game quickly, just double click the file (.nds) and select "open with melonds" if it asks. It should only ask the first time you try to open a game, and then it works for all games. You can even create shortcuts to the files and give them custom icons.

thats just a normal way to open games , the suggestion is for closing melonds , coming back at any future point , opening it back up , and its right where you left off , similar to playing a game on your actual ds , closing it , coming back at any future point , and opening it back up and having it be right where you left off
